independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/inheritance/TypeLevelInheritanceTest.java

```diff
@@ -0,0 +1,183 @@
+            .build();
+
+    @Test
+    public void testTypeInheritance() {
+        ArcContainer container = Arc.container();
+
+        assertEquals(0, FirstInterceptor.timesInvoked);
+        assertEquals(0, SecondInterceptor.timesInvoked);
+        assertEquals(0, ThirdInterceptor.timesInvoked);
+
+        BasicBean basicBean = container.instance(BasicBean.class, new MyQualifier.Literal(), new MyQualifier2.Literal()).get();
+        assertNotNull(basicBean);
+        basicBean.ping();
+        assertEquals(1, FirstInterceptor.timesInvoked);
+        assertEquals(1, SecondInterceptor.timesInvoked);
+        assertEquals(1, ThirdInterceptor.timesInvoked);
+
+        SubBean subBean = container.instance(SubBean.class, new MyQualifier.Literal(), new MyQualifier2.Literal()).get();
+        assertNull(subBean);
+        subBean = container.instance(SubBean.class, new MyQualifier.Literal()).get();
+        assertNotNull(subBean);
+        subBean.ping();
+        assertEquals(2, FirstInterceptor.timesInvoked);
+        assertEquals(2, SecondInterceptor.timesInvoked);
+        assertEquals(1, ThirdInterceptor.timesInvoked);
+    }
+
+    @ApplicationScoped
+    @MyBinding
+    @MyQualifier
+    @MyQualifier2
+    @MyStereotype2
+    @MyStereotype
+    static class BasicBean {
+        public void ping() {
+
+        }
+    }
+
+    @Typed(SubBean.class)
+    @ApplicationScoped
+    static class SubBean extends BasicBean {
+
+    }
+
+    @MyBinding
+    @Interceptor
+    @Priority(1)
+    static class FirstInterceptor {
+        public static int timesInvoked = 0;
+
+        @AroundInvoke
+        Object aroundInvoke(InvocationContext ctx) throws Exception {
+            timesInvoked++;
+            return ctx.proceed();
+        }
+    }
+
+    @ThirdBinding
+    @Interceptor
+    @Priority(2)
+    static class ThirdInterceptor {
+        public static int timesInvoked = 0;
+
+        @AroundInvoke
+        Object aroundInvoke(InvocationContext ctx) throws Exception {
+            timesInvoked++;
+            return ctx.proceed();
+        }
+    }
+
+    @SecondaryBinding
+    @Interceptor
+    @Priority(1)
+    static class SecondInterceptor {
+        public static int timesInvoked = 0;
+
+        @AroundInvoke
+        Object aroundInvoke(InvocationContext ctx) throws Exception {
+            timesInvoked++;
+            return ctx.proceed();
+        }
+    }
+
+    @Target({ TYPE, METHOD, FIELD, PARAMETER })
+    @Retention(RUNTIME)
+    @Inherited
+    @InterceptorBinding
+    @interface MyBinding {
+    }
+
+    @Stereotype
+    @Inherited
+    @SecondaryBinding
+    @Target({ TYPE, METHOD, FIELD, PARAMETER })
+    @Retention(RUNTIME)
+    @interface MyStereotype {
+    }
+
+    @Stereotype
+    @ThirdBinding
+    @Target({ TYPE, METHOD, FIELD, PARAMETER })
+    @Retention(RUNTIME)
+    @interface MyStereotype2 {
+    }
+
+    @Target({ TYPE, METHOD, FIELD, PARAMETER })
+    @Retention(RUNTIME)
+    @Inherited
```

Comment:
This doesn't have to be `@Inherited`, right? It's applied to `SubBean`, because that inherits `@MyStereotype`.

Reply:
I was thinking if it's possible this could mask an issue, but probably not.

Reply:
fixed, I have removed it @Ladicek  @manovotn